src/useCapabilitiesSafe.ts

```diff
@@ -0,0 +1,36 @@
+import type { Chain } from 'viem';
+import { useAccount } from 'wagmi';
+import { useCapabilities } from 'wagmi/experimental';
+import type { WalletCapabilities } from './types';
+
+export function useCapabilitiesSafe({
+  chain,
+}: {
+  chain: Chain;
+}): WalletCapabilities {
+  const { connector, isConnected } = useAccount();
+
+  // Metamask doesn't support wallet_getCapabilities
+  const isMetamaskWallet = connector?.id === 'io.metamask';
+  const enabled = isConnected && !isMetamaskWallet;
+
+  const { data: capabilities } = useCapabilities({ query: { enabled } });
+
+  const atomicBatchEnabled =
```

Comment:
I think we can simplify the readability of this code as

```ts
let hasAtomicBatch = false;
let hasPaymasterService = false;
let hasAuxiliaryFunds= false;

if (capabilities &&  capabilities[chain.id]) {
  const capabilitiesByChain = capabilities[chain.id];
  hasAtomicBatch = capabilitiesByChain.atomicBatch?.supported  === true;
  hasPaymasterService = capabilitiesByChain.paymasterService?.supported  === true;
  hasAuxiliaryFunds = capabilitiesByChain.auxiliaryFunds?.supported === true;
}
```

or something like that. Just avoid all the crazy tuples, that are always hard to mantain.

Reply:
i just copy pasted the Basenames code for `useCapabilitiesSafe`, just edited to make it more succinct